src/clj_kondo/impl/analyzer/re_frame.clj

```diff
@@ -40,7 +50,46 @@
           (common/analyze-children ctx args)))
       (common/analyze-children ctx children))))
 
-(defn analyze-reg-sub [ctx expr fq-def]
+(defn analyze-inject-cofx
+  "Analyzes re-frame co effect injection.
+  Marks the co effect id, the first parameter as a co effect reference so it can be tracked back
+  to the co effect registration. Marker is `[:context :re-frame.core :cofx-ref]`.
+  https://day8.github.io/re-frame/api-re-frame.core/#inject-cofx"
+  [ctx expr ns]
+  (let [kns (keyword ns)
+        [cofx-id & cofx-params] (next (:children expr))]
+    (common/analyze-children (assoc-in ctx [:context kns :cofx-ref] true) [cofx-id])
+    (when cofx-params
+      (common/analyze-children ctx cofx-params))))
+
+(defn- analyze-dispatch-event-id
+  "Marks the event id, the first element of the parameter vector as an event reference so
+  it can be tracked back to the event registration. Marker is `[:context :re-frame.core :event-ref]`.
+  https://day8.github.io/re-frame/api-re-frame.core/#dispatch"
+  [ctx expr ns]
+  (let [kns (keyword ns)
+        [farg & _args :as _children] (next (:children expr))]
+    (when (identical? :vector (utils/tag farg))
+      (let [[event-id & _event-params] (:children farg)]
+        (common/analyze-children (assoc-in ctx [:context kns :event-ref] true) [event-id])))))
+
+(defn analyze-dispatch
+  "Analyzes re-frame dispatch call.
+  Uses [[analyze-dispatch-event-id]] to analyze the event id. And analyzes the rest of the dispatch call
+  as usual."
+  [ctx expr ns]
+  (analyze-dispatch-event-id ctx expr ns)
+  (let [[name-expr & body] (next (:children expr))]
+    (common/analyze-children ctx (cons name-expr body))))
```

Comment:
Why are the args here destructured but then cons-ed again?

Reply:
lazy refactor on my side. fixing...

Reply:
done